Libraries/LibWeb/WebGL/WebGL2RenderingContextImpl.cpp

```diff
@@ -3078,6 +3078,31 @@ Optional<String> WebGL2RenderingContextImpl::get_shader_info_log(GC::Root<WebGLS
     return String::from_utf8_without_validation(ReadonlyBytes { info_log.data(), static_cast<size_t>(info_log_length - 1) });
 }
 
+Optional<String> WebGL2RenderingContextImpl::get_shader_source(GC::Root<WebGLShader> shader)
+{
+    m_context->make_current();
+
+    GLuint shader_handle = 0;
+    if (shader) {
+        auto handle_or_error = shader->handle(this);
+        if (handle_or_error.is_error()) {
+            set_error(GL_INVALID_OPERATION);
+            return {};
+        }
+        shader_handle = handle_or_error.release_value();
+    }
+
+    GLint shader_source_length = 0;
+    glGetShaderiv(shader_handle, GL_SHADER_SOURCE_LENGTH, &shader_source_length);
+    if (!shader_source_length)
+        return String {};
+
+    Vector<GLchar> shader_source;
+    shader_source.resize(shader_source_length);
+    glGetShaderSource(shader_handle, shader_source_length, nullptr, shader_source.data());
```

Comment:
Using `ByteBuffer` instead of `Vector` might be the more idiomatic way to have a place to put bytes in.